diff --git a/test/distributed/org/apache/cassandra/distributed/test/metric/TableMetricTest.java b/test/distributed/org/apache/cassandra/distributed/test/metric/TableMetricTest.java
index 58a3feef40..e4c3f967d3 100644
--- a/test/distributed/org/apache/cassandra/distributed/test/metric/TableMetricTest.java
+++ b/test/distributed/org/apache/cassandra/distributed/test/metric/TableMetricTest.java
@@ -21,6 +21,7 @@ package org.apache.cassandra.distributed.test.metric;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
@@ -277,7 +278,7 @@ public class TableMetricTest extends TestBaseImpl
         @Override
         public Set<ObjectName> queryNames(ObjectName name, QueryExp query)
         {
-            return null;
+            return Collections.emptySet();
         }

         @Override
